app/controllers/application_controller/ci_processing.rb

```diff
@@ -1868,6 +1868,7 @@ def process_cloud_object_storage_buttons(pressed)
     task = pressed.sub("#{klass.name.underscore.to_sym}_", "")
 
     return tag(klass) if task == "tag"
+    return javascript_redirect(:controller => klass.name.underscore.to_sym, :action => "new") if task == "new"
 
```

Comment:
Sounds reasonable. Let me try to do it this way. What confuses me still is this line in `ems_storage` controller: https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/ems_common.rb#L285. It seems like delete for `cloud_volume` is implemented in ci_processing while other operations are elsewhere. Let me try to refactor this, we shall see if any problem arises.